src/main/java/com/microsoft/jenkins/azuread/integrations/casc/AzureAdMatrixAuthorizationStrategyConfigurator.java

```diff
@@ -0,0 +1,110 @@
+package com.microsoft.jenkins.azuread.integrations.casc;
+
+import com.microsoft.jenkins.azuread.AzureAdMatrixAuthorizationStrategy;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.security.AuthorizationStrategy;
+import hudson.security.Permission;
+import io.jenkins.plugins.casc.Attribute;
+import io.jenkins.plugins.casc.BaseConfigurator;
+import io.jenkins.plugins.casc.ConfigurationContext;
+import io.jenkins.plugins.casc.impl.attributes.MultivaluedAttribute;
+import io.jenkins.plugins.casc.model.CNode;
+import io.jenkins.plugins.casc.model.Mapping;
+import org.jenkinsci.plugins.matrixauth.AuthorizationContainer;
+import org.jenkinsci.plugins.matrixauth.integrations.PermissionFinder;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+@Extension(optional = true, ordinal = 2)
+@Restricted(NoExternalUse.class)
+public class AzureAdMatrixAuthorizationStrategyConfigurator extends
+        BaseConfigurator<AzureAdMatrixAuthorizationStrategy> {
+    @NonNull
+    @Override
+    public String getName() {
+        return "azureAdMatrix";
+    }
+
+    @Override
+    public Class<AzureAdMatrixAuthorizationStrategy> getTarget() {
+        return AzureAdMatrixAuthorizationStrategy.class;
+    }
+
+
+    @NonNull
+    @Override
+    public Class getImplementedAPI() {
+        return AuthorizationStrategy.class;
+    }
+
+    @Override
+    @NonNull
+    public Set<Attribute<AzureAdMatrixAuthorizationStrategy, ?>> describe() {
+        return new HashSet<>(Arrays.asList(
+                new MultivaluedAttribute<AzureAdMatrixAuthorizationStrategy, String>("permissions", String.class)
+                        .getter(AzureAdMatrixAuthorizationStrategyConfigurator::getPermissions)
+                        .setter(AzureAdMatrixAuthorizationStrategyConfigurator::setPermissions),
+
+                // support old style configuration options
```

Comment:
no need to support `grantedPermissions`
```suggestion
```
this block can be deleted from here on